### PR TITLE
ParseCountryCode method implemented.

### PR DESCRIPTION
**Pull request by VlastimirAdlerIII**
ParseCountryCode method implemented to replace repetitive code. Some minor name tuning. Few tests added.

### CODE REVIEW

File: src/Nager.Date/DateSystem.cs [hunk 1]
```diff
@@ -185,6 +187,21 @@ public static IPublicHolidayProvider GetProvider(CountryCode countryCode)
             return GetPublicHolidayProvider(countryCode);
         }
 
+        /// <summary>
+        /// GetPublicHolidayProvider
+        /// </summary>
+        /// <param name="countryCode">Country Code (ISO 3166-1 ALPHA-2)</param>
+        /// <returns></returns>
```
Comment (samtrion) [inline src/Nager.Date/DateSystem.cs:194]: Please describe the return value.

File: src/Nager.Date/DateSystem.cs [hunk 2]
```diff
@@ -200,6 +217,21 @@ public static IPublicHolidayProvider GetPublicHolidayProvider(CountryCode countr
             return NoHolidaysProvider.Instance;
         }
 
+        /// <summary>
+        /// GetWeekendProvider
+        /// </summary>
+        /// <param name="countryCode">Country Code (ISO 3166-1 ALPHA-2)</param>
+        /// <returns></returns>
```
Comment (samtrion) [inline src/Nager.Date/DateSystem.cs:224]: Please describe the return value.

File: src/Nager.Date/DateSystem.cs [hunk 3]
```diff
@@ -215,6 +247,17 @@ public static IWeekendProvider GetWeekendProvider(CountryCode countryCode)
             return WeekendProvider.Universal;
         }
 
+        /// <summary>
+        /// Parses given string to find CurrencyCode
+        /// </summary>
+        /// <param name="countryCode"></param>
+        /// <param name="parsedCountryCode"></param>
+        /// <returns></returns>
```
Comment (samtrion) [inline src/Nager.Date/DateSystem.cs:255]: Please describe the return value.

File: src/Nager.Date/DateSystem.cs [hunk 4]
```diff
@@ -516,6 +559,22 @@ public static bool IsOfficialPublicHolidayByCounty(DateTime date, CountryCode co
 
         #region Check a date is a Weekend
 
+        /// <summary>
+        /// Check is a give date in the Weekend
+        /// </summary>
+        /// <param name="date">The date to check</param>
+        /// <param name="countryCode">Country Code (ISO 3166-1 ALPHA-2)</param>
+        /// <returns></returns>
```
Comment (samtrion) [inline src/Nager.Date/DateSystem.cs:567]: Please describe the return value.

File: src/Nager.Date/DateSystem.cs [hunk 5]
```diff
@@ -516,6 +559,22 @@ public static bool IsOfficialPublicHolidayByCounty(DateTime date, CountryCode co
...
+        public static bool IsWeekend(DateTime date, string countryCode)
+        {
+            if (!ParseCountryCode(countryCode, out var parsedCountryCode))
+            {
+                throw new ArgumentException(string.Format(CountryCodeParsingError, countryCode));
```
Comment (samtrion) [inline src/Nager.Date/DateSystem.cs:572]: Please document the ArgumentException in the XML Summary, to complete the Public API.

File: src/Nager.Date/DateSystem.cs [hunk 6]
```diff
@@ -200,6 +217,21 @@ public static IPublicHolidayProvider GetPublicHolidayProvider(CountryCode countr
...
+        /// <summary>
+        /// GetWeekendProvider
+        /// </summary>
+        /// <param name="countryCode">Country Code (ISO 3166-1 ALPHA-2)</param>
+        /// <returns></returns>
+        public static IWeekendProvider GetWeekendProvider(string countryCode)
+        {
+            if (!ParseCountryCode(countryCode, out var parsedCountryCode))
+            {
+                throw new ArgumentException(string.Format(CountryCodeParsingError, countryCode));
```
Comment (samtrion) [inline src/Nager.Date/DateSystem.cs:229]: Please document the ArgumentException in the XML Summary, to complete the Public API.

File: src/Nager.Date/DateSystem.cs [hunk 7]
```diff
@@ -185,6 +187,21 @@ public static IPublicHolidayProvider GetProvider(CountryCode countryCode)
...
+        /// <summary>
+        /// GetPublicHolidayProvider
+        /// </summary>
+        /// <param name="countryCode">Country Code (ISO 3166-1 ALPHA-2)</param>
+        /// <returns></returns>
+        public static IPublicHolidayProvider GetPublicHolidayProvider(string countryCode)
+        {
+            if (!ParseCountryCode(countryCode, out var parsedCountryCode))
+            {
+                throw new ArgumentException(string.Format(CountryCodeParsingError, countryCode));
```
Comment (samtrion) [inline src/Nager.Date/DateSystem.cs:199]: Please document the ArgumentException in the XML Summary, to complete the Public API.

File: src/Nager.Date/DateSystem.cs (hunk 1 above)
Comment (VlastimirAdlerIII) [reply src/Nager.Date/DateSystem.cs]: This was not only place where return info was missing, so I tried to find all such places and add some description.
(Obsolete members omited.)
Same for exceptions.

**Review by samtrion — APPROVED** [review]

**Pull request merged**